csml_engine/src/data.rs

```diff
@@ -93,6 +93,173 @@ impl BotOpt {
     }
 }
 
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SerializeCsmlBot {
+    pub id: String,
+    pub name: String,
+    pub flows: Vec<CsmlFlow>,
+    pub native_components: Option<String>, // serde_json::Map<String, serde_json::Value>
+    pub custom_components: Option<String>, // serde_json::Value
+    pub default_flow: String,
+    pub env: Option<String>,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
```

Comment:
Add some comments why this is necessary! In a few months there is no way anyone understands why this is there.

Also, TMPSerializeCsmlBot looks temporary. Either this requires a mention about why it is temporary (and when we can remove it) or we need to change the name to something more suitable like SerializeBincodeEncodedCsmlBot